### PR TITLE
Handle files renamed in `get_file_modifications`

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?
Handle files renamed in `get_file_modifications`.

`git diff --name-status` displays 3 parts when a file was renamed.
Example:
```
['R100', 'comp/core/tagger/taggerimpl/api/getlist.go', 'comp/core/tagger/api/getlist.go']
```
### Describe how to test/QA your changes
QA done by running unit tests.

### CODE REVIEW

**Review by chouetz — COMMENTED** [review]
I'm wondering if we could/should just change the default value of the `no_renames` to `True` instead, wdyt?

File: tasks/libs/common/git.py [hunk 1]
```diff
@@ -71,17 +71,23 @@ def get_file_modifications(
...
+        modifications = []
+        for m in raw_modifications:
+            if len(m) == 3:
+                # file was renamed
+                continue
+            if len(m) != 2:
+                raise ValueError(f"Unexpected git diff output: {m}")
+
+            status = m[0]
+            file = m[1]
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply tasks/libs/common/git.py]: Code updated

**Review by CelianR — APPROVED** [review]

**Review by pducolin — APPROVED** [review]

File: tasks/libs/common/git.py [hunk 2]
```diff
@@ -71,17 +71,19 @@ def get_file_modifications(
...
+        for m in raw_modifications:
+            if len(m) == 3:
+                # file was renamed
+                continue
+            status, file = m
+            if (added and status == "A") or (modified and status in "MCRT") or (removed and status == "D"):
+                modifications.append((status, file))
 
     if only_names:
         modifications = [file for _, file in modifications]
```
Comment (pducolin) [inline tasks/libs/common/git.py:89]: **💭 thought**
Is it fine to have this change of the logic here? Before this was taking current `raw_modifications` when `not added and not modified and not removed`, now it's taking a non defined array

Comment (ogaca-dd) [reply tasks/libs/common/git.py]: Good catch. Code updated.

**Review by pducolin — APPROVED** [review]

**Pull request merged**